src/test/tst_attributerange.cpp

```diff
@@ -30,10 +31,628 @@ class TestAttributeRange: public QObject
+    else QVERIFY(v.isValid());
+
+    QCOMPARE(t, type);
+
+    switch (type) {
+    case Value::BOOL:
+        QVERIFY(v.isBool());
+        QVERIFY(!v.isInt());
+        QVERIFY(!v.isChar());
+        QVERIFY(!v.isDouble());
+        QVERIFY(!v.isString());
+        break;
+    case Value::INT:
+        QVERIFY(v.isInt());
+        QVERIFY(!v.isBool());
+        QVERIFY(!v.isChar());
+        QVERIFY(!v.isDouble());
+        QVERIFY(!v.isString());
+        break;
+    case Value::DOUBLE:
+        QVERIFY(v.isDouble());
+        QVERIFY(!v.isInt());
+        QVERIFY(!v.isBool());
+        QVERIFY(!v.isChar());
+        QVERIFY(!v.isString());
+        break;
+    case Value::STRING:
+        QVERIFY(v.isString());
+        QVERIFY(!v.isInt());
+        QVERIFY(!v.isBool());
+        QVERIFY(!v.isDouble());
+        QVERIFY(!v.isChar());
+        break;
+    default:
+        QVERIFY(!v.isString());
+        QVERIFY(!v.isInt());
+        QVERIFY(!v.isBool());
+        QVERIFY(!v.isDouble());
+        QVERIFY(!v.isChar());
+        break;
+    }
+}
+
+void TestAttributeRange::tst_bool()
+{
+    AttributeRange* attrRge = AttributeRange::parse(0, "test", "bool");
+
+    // Tests value returned by 'AttributeRange::validate()' for true
+    Value v;
+
+    v = attrRge->validate("true");
+    _tst_value(v, Value::BOOL);
+
+    // Tests value returned by 'AttributeRange::validate()' for false
+    v = attrRge->validate("false");
+    _tst_value(v, Value::BOOL);
+
+    // Tests value returned by 'AttributeRange::validate()' for an invalid string
+     v = attrRge->validate("invalid");
+     _tst_value(v, Value::INVALID);
+
+    // Tests if functions work as expected
+    QVERIFY(attrRge->isValid());
+    QCOMPARE(attrRge->id(), 0);
+    QCOMPARE(attrRge->attrName(), "test");
+    QCOMPARE(attrRge->attrRangeStr(), "bool");
+
+    AttributeRange::Type type = AttributeRange::Bool;
+    QCOMPARE(attrRge->type(), type);
+
+    // Tests min(), max() and rand() functions work as expected
+    // where min and max are boolean values
+    bool min_bool = false;
+    bool max_bool = true;
+
+    // where min and max are integers, casted to boolean type
+    bool min_int = Value(0).toBool();
+    bool max_int = Value(1).toBool();
+
+    // where min and max are booleans, casted to integer type
+    int min_int_alt = Value(false).toInt();
+    int max_int_alt = Value(true).toInt();
+
+    // For a boolean type, 'AttributeRange::rand()' returns a bool
+    PRG* prg = new PRG(123);
+    v = attrRge->rand(prg);
+
+    _tst_value(v, Value::BOOL);
+
+    QVERIFY(v >= min_bool);
+    QVERIFY(v <= max_bool);
+    QVERIFY(v >= min_int);
+    QVERIFY(v <= max_int);
+    QVERIFY(v.toInt() >= min_int_alt);
+    QVERIFY(v.toInt() <= max_int_alt);
+
+    // For a boolean type, 'AttributeRange::min()' and 'AttributeRange::max()'
+    // both return <null>
+//  QCOMPARE(attrRge->min(), min_bool);  // Succeeds, as <null> is recognised as 0
```

Comment:
they fail because it's comparing a `Value` with a `bool`. It should work:
``` c++
QCOMPARE(attrRge->min().toBool(), min_bool);
```
of course, changing the defininition of `bool min_bool = false;` to  `Value min_bool = false;` should also work. But I think it's better to use `.toBool()` solution.

Reply:
Thank you for the feedback. Something I noticed while testing this class is that I missed this situation when writing the tests for the Value class. Is this something that would be a good idea to include in that test? (This particular one fails)
```c++
    Value v1 = Value(false);
    QCOMPARE(v1.toInt(), 0);

Reply:
In this case, as you're defining a `Value` with a `boolean`, reading it with anything else than `toBool` should fail.